caikit/interfaces/vision/data_model/backends/image_pil_backend.py

```diff
@@ -0,0 +1,174 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Standard
+from typing import Any, Union
+import io
+import os
+
+# Third Party
+from PIL import Image as PILImage
+import numpy as np
+
+# First Party
+import alog
+
+# Local
+from caikit.core import error_handler
+from caikit.core.data_model.base import DataBase
+from caikit.core.data_model.data_backends import DataModelBackendBase
+
+log = alog.use_channel("DATABACK")
+error = error_handler.get(log)
+
+
+class ImagePilBackend(DataModelBackendBase):
+    def __init__(self, image_data):
+        self._image_data = self.__class__.coerce_to_pil(image_data)
+        # Unless the PIL image explicitly has information about its format, export
+        # serialized image data as a PNG since it's not a lossy compression
+        self._export_format = (
+            self._image_data.format if self._image_data.format is not None else "PNG"
+        )
+
+    def get_attribute(self, data_model_class: DataBase, name: str) -> Any:
+        # Convert the underlying PIL image to bytes
+        if name == "image_data":
+            # Save the whole contents of the image into a BytesIO object; this includes image fmt
+            im_arr = io.BytesIO()
+            self._image_data.save(im_arr, format=self._export_format)
+            # Then return the whole value of the buffer as a bytes object
+            return im_arr.getvalue()
+        # Delegate to common parent logic
+        return super().get_attribute(data_model_class, name)
+
+    def cache_attribute(self, name: str, value: Any) -> bool:
```

Comment:
Sounds good - deleting this and letting it fall back to the default `true` then, since `image_data` is currently our only attribute on the protobuf anyway 😄